tests/test_remote_source_and_cache.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import os
+import shutil
+
+import geopandas
+import intake
+import pytest
+from fiona.errors import DriverError
+
+from intake_geopandas import GeoJSONSource, ShapefileSource
+
+
+def try_clean_old_cache(item):
+    if isinstance(item.cache, list):
+        if len(item.cache) > 0:
+            if isinstance(item.cache[0], str):
+                item.cache[0].clear_cache()
+
+
+def try_clean_new_cache(item):
+    c = None  # default
+    for c in ['blockcache', 'filecache', 'simplecache']:
+        if c in item.storage_options:
+            caching = c
+    assert c is not None, 'caching not found'
+    path = item.storage_options[caching]['cache_storage']
+    if isinstance(path, str):
+        if os.path.exists(path):
+            shutil.rmtree(path)
```

Comment:
would `fsspec.clear_cache()` also work?

Reply:
`clear_cache` is a method on the file-system instance (because the location of the files is configurable), so you can only call it if you've made the instance. It, in turn, calls rmtree anyway, so this is fine.